gems/aws-crt-auth/lib/aws-crt-auth/signer.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Aws
+  module Crt
+    module Auth
+      # Utility class for creating AWS signature version 4 signature.
+      class Signer
```

Comment:
So, in all the other CRTs, AwsSigningConfig was a class that strictly held data.

The signing functions were either module functions that belonged to no class (python), or for java they were static functions on an AwsSigner class that existed solely to host static functions.

I can't say you're wrong, putting data in the same class as the functions which act on that data. That's the textbook definition of OOP... but something about this being different from the other CRTs make me uneasy.

Reply:
To be honest this isn't really meant to be a CRT interface (the gem is still in the CRT repo for now because it makes development easier, but would eventually be moved out to the SDK) - this class is intended to provide an identical interface to the existing SDK Signer class so that it can be a drop in replacement.   See the next PR: #21  - it seperates this SDK interface out into its own gem + creates a SigningConfig class that is strictly data and a static sign_request function that operates on signable + signing config.

Reply:
Though it's also worth debating on that next PR whether we want to create some more intermediate classes/functions for signing. The SDK signer (and any other code that used the CRT bindings has to have a fairly Low level knowledge of the interface, eg callbacks and property lists, ect)